### PR TITLE
Bump symfony version to 2.8

### PR DESCRIPTION
**Pull request by core23**
I am targetting this branch, because this is a major change.

## Changelog

``` markdown
### Changed
- Bumped Symfony version to 2.8
```
## To do
- [x] Setup dev-kit https://github.com/sonata-project/dev-kit/pull/154
## Subject

Only the latest LTS release is supported for the next major release.

### CODE REVIEW

**Comment by Koc** [comment]
so maybe you can here cleanup code for hacks for Symfony < 2.8

**Comment by OskarStark** [comment]
Please write `Symfony` instead of `symfony` in commit msg

**Comment by greg0ire** [comment]
793 deletions :)

before
after

File: Form/Type/BaseStatusType.php [hunk 1]
```diff
@@ -56,11 +56,7 @@ public function __construct($class, $getter, $name, $flip = false)
      */
     public function getParent()
     {
-        // NEXT_MAJOR: Remove ternary and keep 'Symfony\Component\Form\Extension\Core\Type\ChoiceType'
-        // (when requirement of Symfony is >= 2.8)
-        return method_exists('Symfony\Component\Form\AbstractType', 'getBlockPrefix')
-            ? 'Symfony\Component\Form\Extension\Core\Type\ChoiceType'
-            : 'choice';
+        return 'Symfony\Component\Form\Extension\Core\Type\ChoiceType';
```
Comment (Koc) [inline Form/Type/BaseStatusType.php]: Consider bumping php version to 5.5 or 5.6 also and use `ChoiceType::class` notation

File: Form/Type/BooleanType.php [hunk 2]
```diff
@@ -74,14 +64,11 @@ public function configureOptions(OptionsResolver $resolver)
             },
         );
 
-        // SF 2.7+ BC
-        if (method_exists('Symfony\Component\Form\AbstractType', 'configureOptions')) {
-            $choices = array_flip($choices);
+        $choices = array_flip($choices);
```
Comment (Koc) [inline Form/Type/BooleanType.php]: better flip by hand in array declaration and remove function call

File: Form/Type/BaseStatusType.php (hunk 1 above)
Comment (core23) [inline Form/Type/BaseStatusType.php]: Hope when can automate this with https://github.com/FriendsOfPHP/PHP-CS-Fixer/issues/1975

**Comment by Soullivaneuh** [comment]
Great job @core23 

Will be mergeable when dev-kit will be updated, as for the other.

I'd suggest you to wait this before making other PR of this kind, because you might have to rebase this PR regularly to fix conflicts from 3.x.

**Comment by core23** [comment]
Do you have any plans when this will happen? @Soullivaneuh

**Comment by greg0ire** [comment]
Let's try a new build.

File: Form/Type/BooleanType.php [hunk 3]
```diff
@@ -36,39 +35,26 @@ public function buildForm(FormBuilderInterface $builder, array $options)
...
 
         $defaultOptions = array(
             'transform' => false,
             'translation_domain' => 'SonataCoreBundle',
         );
 
-        // NEXT_MAJOR: Remove this "if" (when requirement of Symfony is >= 2.7)
-        if (method_exists('Symfony\Component\Form\AbstractType', 'configureOptions')) {
-            $choices = array_flip($choices);
+        $choices = array_flip($choices);
```
Comment (jordisala1991) [inline Form/Type/BooleanType.php]: Something is strange here... if you flip the choices every time, wouldn't be better to just flip them in the array creation just above?

Also, if we are always setting them on the `$defaultOptions['choices']`, IMO it is better to just declare them on that array.

**Comment by OskarStark** [comment]
shall we merge this @core23 @greg0ire ?

**Comment by greg0ire** [comment]
Yes! Thanks a lot @core23 !

**Pull request merged**